pytorch_lightning/core/lightning.py

```diff
@@ -163,6 +164,7 @@ def example_input_array(self, example: Any) -> None:
         self._example_input_array = example
 
     @property
+    @deprecated(deprecated_in="1.3.0", remove_in="1.5.0", target=None)
```

Comment:
I think it's better to use `rank_zero_warn` here as there is not target.

Also for the setter

cc: @Borda